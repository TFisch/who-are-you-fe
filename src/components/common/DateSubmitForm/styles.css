.date-submit-form {
  align-items: center;
  border: 1px solid black;
  display: flex;
  flex-direction: column;
  height: 23em;
  width: 40%; }
