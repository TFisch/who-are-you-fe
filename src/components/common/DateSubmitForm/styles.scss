.date-submit-form {
  align-items: center;
  border: 1px solid black;
  border-radius: 19rem;
  display: flex;
  flex-direction: column;
  height: 25em;
  width: 40%;
  margin: 5rem auto auto auto;
  background-color: rgba(235, 244, 244, 0.8);
}

.form-title {
  margin-top: 2.5rem;
}

.error {
  color: crimson;
}
