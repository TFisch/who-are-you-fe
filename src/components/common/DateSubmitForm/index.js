import React, { Component } from 'react';
import Submit from '../Submit';
import './styles.scss';
import { concatDates, getMonthString, checkDay, validateBirthday } from '../../../utilities/helper';

class DateSubmitForm extends Component {
  constructor() {
    super();
    this.state = {
      nameInput: "",
      dateInput: 0,
      dateSubmitted: false,
      dobError: ""
    }

  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };


  handleSubmit = (e) => {
    e.preventDefault();
    const date = this.state.dateInput;
    const day = date.substring(8, 10);
    const month = date.substring(5, 7);
    const year = date.substring(0, 4);
    const checkBirthday = validateBirthday(day, month, year);
    if (checkBirthday === "error") {
      this.setState({ dobError: "Sorry! Birthday must fall between 1970 and 1996" })
    } else {
      const dayString = checkDay(day);
      const monthString = getMonthString(month);
      const cleanedDate = concatDates(monthString, dayString);
      this.setState({ dateSubmitted: true });
      this.props.hideForm();
    }
  }


  render() {

    return (
      <form className="date-submit-form">
        <h1>{this.props.headerText}</h1>
        <h3>{this.props.inputOneText}</h3>
        <input
          type="text"
          name="nameInput"
          onChange={this.handleChange}
          value={this.nameInput}
        />
        <h3>{this.props.inputTwoText}</h3>
        <input type="date" name="dateInput" min="2000-01-02" max="2004-12-31" onChange={this.handleChange} value={this.dateInput} />
        <p>{this.state.dobError}</p>
        <Submit handleSubmit={this.handleSubmit} buttonText="SUBMIT" />

      </form>
    );
  }
}

export default DateSubmitForm;
