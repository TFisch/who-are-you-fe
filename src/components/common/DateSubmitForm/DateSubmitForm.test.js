import React from 'react';
import { shallow } from 'enzyme';
import DateSubmitForm from './'

describe('DateSubmitForm', () => {
  let wrapper;

  beforeEach(() => {
    const mockHideForm = jest.fn()
    let mockProps = {
      headerText: "WHO ARE YOU?",
      inputOneText: "What is your name?",
      inputTwoText: "What is your birthday?",
      hideForm: mockHideForm
    }

    wrapper = shallow(<DateSubmitForm props={mockProps}
    />);
  })

  it('should be a test', () => {

  });
});