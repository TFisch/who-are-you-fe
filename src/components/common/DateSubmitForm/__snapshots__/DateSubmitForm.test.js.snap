// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSubmitForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DateSubmitForm
    props={
      Object {
        "headerText": "WHO ARE YOU?",
        "hideForm": [MockFunction],
        "inputOneText": "What is your name?",
        "inputTwoText": "What is your birthday?",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <h3 />,
        <input
          name="nameInput"
          onChange={[Function]}
          type="text"
        />,
        <p>
          
        </p>,
        <h3 />,
        <input
          max="2004-12-31"
          min="2000-01-02"
          name="dateInput"
          onChange={[Function]}
          type="date"
        />,
        <p>
          
        </p>,
        <Submit
          buttonText="SUBMIT"
          handleSubmit={[Function]}
        />,
      ],
      "className": "date-submit-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "nameInput",
          "onChange": [Function],
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "max": "2004-12-31",
          "min": "2000-01-02",
          "name": "dateInput",
          "onChange": [Function],
          "type": "date",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "SUBMIT",
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <h3 />,
          <input
            name="nameInput"
            onChange={[Function]}
            type="text"
          />,
          <p>
            
          </p>,
          <h3 />,
          <input
            max="2004-12-31"
            min="2000-01-02"
            name="dateInput"
            onChange={[Function]}
            type="date"
          />,
          <p>
            
          </p>,
          <Submit
            buttonText="SUBMIT"
            handleSubmit={[Function]}
          />,
        ],
        "className": "date-submit-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "nameInput",
            "onChange": [Function],
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "max": "2004-12-31",
            "min": "2000-01-02",
            "name": "dateInput",
            "onChange": [Function],
            "type": "date",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "SUBMIT",
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
